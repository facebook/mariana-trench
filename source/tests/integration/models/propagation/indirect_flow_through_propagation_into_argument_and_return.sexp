(method (private) "LFlow;.flow:()V"
 (
  (load-param v0)

  LINE(1)
  (invoke-virtual (v1) "LSource;.source:()LData;")
  (move-result-object v2)

  LINE(2)
  (invoke-virtual (v0 v2) "LFlow;.propagate_argument_into_this:(LSource;)LFlow;")
  (move-result-object v3)

  LINE(3)
  (invoke-virtual (v1 v3) "LSink;.sink:(LData;)V")
  (return-void)
 )
)
(method (private) "LFlow;.propagate_argument_into_this:(LSource;)LFlow;"
 (
  (load-param v0)
  (load-param v1)

  LINE(4)
  (iput-object v1 v0 "LFlow;.field:LSource;")

  LINE(5)
  (return-object v0)
 )
)
