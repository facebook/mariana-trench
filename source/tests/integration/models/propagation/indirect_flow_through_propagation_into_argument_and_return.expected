{
  "metadata" :
  {
    "@generated" : true
  },
  "models" :
  [
    {
      "generations" :
      [
        {
          "call_position" :
          {
            "line" : 1,
            "path" : "Flow.java"
          },
          "callee" : "LSource;.source:()LData;",
          "callee_port" : "Return",
          "caller_port" : "Argument(0).field",
          "distance" : 1,
          "kind" : "TestSource",
          "local_positions" :
          [
            {
              "line" : 2
            }
          ],
          "origins" :
          [
            "LSource;.source:()LData;"
          ]
        }
      ],
      "issues" :
      [
        {
          "position" :
          {
            "line" : 3,
            "path" : "Flow.java"
          },
          "rule" : 1,
          "sinks" :
          [
            {
              "call_position" :
              {
                "line" : 3,
                "path" : "Flow.java"
              },
              "callee" : "LSink;.sink:(LData;)V",
              "callee_port" : "Argument(1)",
              "distance" : 1,
              "kind" : "TestSink",
              "origins" :
              [
                "LSink;.sink:(LData;)V"
              ]
            }
          ],
          "sources" :
          [
            {
              "call_position" :
              {
                "line" : 1,
                "path" : "Flow.java"
              },
              "callee" : "LSource;.source:()LData;",
              "callee_port" : "Return",
              "distance" : 1,
              "kind" : "TestSource",
              "local_positions" :
              [
                {
                  "line" : 2
                }
              ],
              "origins" :
              [
                "LSource;.source:()LData;"
              ]
            }
          ]
        }
      ],
      "method" : "LFlow;.flow:()V",
      "position" :
      {
        "line" : 0,
        "path" : "Flow.java"
      },
      "sinks" :
      [
        {
          "call_position" :
          {
            "line" : 3,
            "path" : "Flow.java"
          },
          "callee" : "LSink;.sink:(LData;)V",
          "callee_port" : "Argument(1)",
          "caller_port" : "Argument(0)",
          "distance" : 1,
          "kind" : "TestSink",
          "local_positions" :
          [
            {
              "line" : 2
            }
          ],
          "origins" :
          [
            "LSink;.sink:(LData;)V"
          ]
        }
      ]
    },
    {
      "method" : "LFlow;.propagate_argument_into_this:(LSource;)LFlow;",
      "position" :
      {
        "line" : 3,
        "path" : "Flow.java"
      },
      "propagation" :
      [
        {
          "input" : "Argument(0)",
          "output" : "Return"
        },
        {
          "input" : "Argument(1)",
          "output" : "Argument(0).field"
        },
        {
          "input" : "Argument(1)",
          "output" : "Return.field"
        }
      ]
    }
  ]
}
